src/main/java/uk/gov/ons/census/fwmt/jobservice/http/rm/RmRestClient.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.ons.census.fwmt.common.data.nc.CaseDetailsDTO;
+import uk.gov.ons.census.fwmt.common.error.GatewayException;
+import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
+import uk.gov.ons.census.fwmt.jobservice.config.CometConfig;
+import uk.gov.ons.census.fwmt.jobservice.http.comet.CometRestClientResponseErrorHandler;
+
+import java.net.MalformedURLException;
+import java.util.Date;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+@Component
+public class RmRestClient {
+
+  private final RestTemplate restTemplate;
+  private final GatewayEventManager gatewayEventManager;
+
+  private final CometConfig cometConfig;
+
+  // temporary store for authentication result
+  private AuthenticationResult auth;
+
+  // derived values
+  private final transient String basePath;
+
+  public static final String FAILED_TM_AUTHENTICATION = "FAILED_TM_AUTHENTICATION";
+
+  public RmRestClient(
+      CometConfig cometConfig,
+      RestTemplateBuilder restTemplateBuilder,
+      GatewayEventManager gatewayEventManager) {
+    this.cometConfig = cometConfig;
+    this.restTemplate = restTemplateBuilder.errorHandler(new CometRestClientResponseErrorHandler())
+        .basicAuthentication(cometConfig.userName, cometConfig.password).build();
+    this.gatewayEventManager = gatewayEventManager;
+    this.basePath = cometConfig.baseUrl + "cases/case-details/";
+    this.auth = null;
+  }
+
+  private boolean isAuthed() {
+    return this.auth != null;
+  }
+
+  private boolean isExpired() {
+    return !auth.getExpiresOnDate().after(new Date());
+  }
+
+  private void auth() throws GatewayException {
+    ExecutorService service = Executors.newFixedThreadPool(1);
+    try {
+      AuthenticationContext context = new AuthenticationContext(cometConfig.authority, false, service);
+      ClientCredential cc = new ClientCredential(cometConfig.clientId, cometConfig.clientSecret);
+
+      Future<AuthenticationResult> future = context.acquireToken(cometConfig.resource, cc, null);
+      this.auth = future.get();
+    } catch (MalformedURLException | InterruptedException | ExecutionException e) {
+      String errorMsg = "Failed to Authenticate with RM Case API";
+      gatewayEventManager
+          .triggerErrorEvent(this.getClass(), errorMsg, "<N/A_CASE_ID>", FAILED_TM_AUTHENTICATION);
+      throw new GatewayException(GatewayException.Fault.SYSTEM_ERROR, errorMsg, e);
+    } finally {
+      service.shutdown();
+    }
+  }
+
+  private HttpHeaders makeAuthHeader() throws GatewayException {
+    if ((!isAuthed() || isExpired()) && !cometConfig.clientId.isEmpty() && !cometConfig.clientSecret.isEmpty())
```

Comment:
No biggie but I'd put this entire line into it's own method returning true or false - i.e boolean needsAuth(); Makes it easier to read.

Reply:
Wasn't needed. This has been removed.